test/infra/fixtures.py

```diff
@@ -2,79 +2,105 @@
 # Licensed under the MIT License.
 
 import os
+from http import HTTPStatus
 from pathlib import Path
-from typing import Optional
+
+import pytest
 
 from pyscitt import governance
 from pyscitt.client import Client
 
 from .did_web_server import DIDWebServer
-from .x5chain_certificate_authority import X5ChainCertificateAuthority
 
+# This file defines a collection of pytest fixtures used to manage and
+# interact with the SCITT ledger.
+
+
+@pytest.fixture(scope="session")
+def service_url():
+    return os.environ.get("CCF_URL", "https://127.0.0.1:8000")
 
-def _get_member_auth():
+
+@pytest.fixture(scope="session")
+def member_auth_path():
     workspace_dir = Path("workspace") / "sandbox_common"
-    cert = workspace_dir.joinpath("member0_cert.pem").read_text()
-    key = workspace_dir.joinpath("member0_privk.pem").read_text()
-    return cert, key
-
-
-class SCITTFixture:
-    service_url: str
-    service_parameters: dict
-    client: Client
-    trust_store: dict
-    did_web_server: DIDWebServer
-    x5c_ca: Optional[X5ChainCertificateAuthority]
-
-    def __init__(
-        self,
-        path: Path,
-        use_default_did_port: bool = False,
-        x5c_ca: X5ChainCertificateAuthority = None,
-    ):
-        self.service_url = os.environ.get("CCF_URL", "https://127.0.0.1:8000")
-        self.client = Client(
-            self.service_url, development=True, member_auth=_get_member_auth()
-        )
+    return (
+        workspace_dir.joinpath("member0_cert.pem"),
+        workspace_dir.joinpath("member0_privk.pem"),
+    )
 
-        self.service_parameters = self.client.get_parameters()
-        self.trust_store = {
-            self.service_parameters["serviceId"]: self.service_parameters,
-        }
-        self.did_web_server = DIDWebServer(
-            use_default_port=use_default_did_port, data_dir=path
-        )
-        self.x5c_ca = x5c_ca
 
-    def __enter__(self):
-        self.did_web_server.__enter__()
+@pytest.fixture(scope="session")
+def member_auth(member_auth_path):
+    return (member_auth_path[0].read_text(), member_auth_path[1].read_text())
+
+
+@pytest.fixture(scope="class")
+def base_client(service_url, member_auth):
+    """
+    Create a Client instance to connect to the test SCITT service.
+
+    Most tests will want to use the `client` fixture instead, which resets the
+    service to a known good state.
+    """
+    return Client(service_url, development=True, member_auth=member_auth)
+
+
+@pytest.fixture(scope="class")
+def configure_service(base_client: Client):
```

Comment:
I think if we rely heavily on fixtures then we should use types more often and check the types since the IDE or mypy won't be able to do it for us.
I found https://github.com/untitaker/pytest-fixture-typecheck which uses https://github.com/agronholm/typeguard. The code of pytest-fixture-typecheck is trivial enough (4 lines of code) to be vendored into our `conftest.py`.

Reply:
I think I'll add the integration with `typeguard` in a separate PR.